model/challenge_results.go

```diff
@@ -0,0 +1,39 @@
+package model
+
+import (
+	"time"
+
+	"google.golang.org/appengine/datastore"
+)
+
+type ChallengeResults []ChallengeResult
+
+func (r ChallengeResults) Len() int {
+	return len(r)
+}
+func (r ChallengeResults) Swap(i, j int) {
+	r[i], r[j] = r[j], r[i]
+}
+func (r ChallengeResults) Less(i, j int) bool {
+	if r[i].TotalTime == 0 {
+		return false
+	}
+	if r[j].TotalTime == 0 {
+		return true
+	}
+	return r[i].TotalTime < r[j].TotalTime
+}
+
+type ChallengeResult struct {
```

Comment:
Not sure whether we should put this here. You are using it only to encode out of a controller, not for persistence like everything else in the model package.